demo/src/features/debug3d/extract.rs

```diff
@@ -1,28 +1,25 @@
-use crate::features::debug3d::plugin::Debug3DStaticResources;
-use crate::features::debug3d::prepare::Debug3dPrepareJobImpl;
-use crate::features::debug3d::{Debug3dRenderFeature, DebugDraw3DResource, ExtractedDebug3dData};
+rafx::declare_render_feature_extract_job!();
+
+use super::{DebugDraw3DResource, StaticResources};
 use rafx::assets::AssetManagerRenderResource;
-use rafx::nodes::{
-    ExtractJob, FramePacket, PrepareJob, RenderFeature, RenderFeatureIndex,
-    RenderJobExtractContext, RenderView,
-};
 
-pub struct Debug3dExtractJob {}
+pub struct ExtractJobImpl {}
 
-impl Debug3dExtractJob {
+impl ExtractJobImpl {
     pub fn new() -> Self {
         Self {}
     }
 }
 
-impl ExtractJob for Debug3dExtractJob {
+impl ExtractJob for ExtractJobImpl {
     fn extract(
         self: Box<Self>,
         extract_context: &RenderJobExtractContext,
         _frame_packet: &FramePacket,
         _views: &[RenderView],
     ) -> Box<dyn PrepareJob> {
-        profiling::scope!("Debug3d Extract");
```

Comment:
I kind of prefer to just use the string directly. Every time we add an indirection to something, it means someone has to go look somewhere else in the code to figure out what's going on, and the value indirecting here brings is minimal.

Reply:
The benefit of the indirection is that this is just boilerplate that needs to be changed every time you work on a feature. It's easy to forget it, or to not keep it up to date if we change the format of it in same way. With this short-hand, it's N less things to need to remember to change when you copy / paste a new feature.

As for needing to look elsewhere in the code -- if you have any type of editor or IDE that desugars Rust (e.g. for type inference), this shouldn't be hidden at all.

![image](https://user-images.githubusercontent.com/4888530/114217848-918d7400-991d-11eb-870c-925adaab1fd3.png)

Reply:
I updated the comment on the `declare_render_feature` macro so that it'll be recognized as an actual doc comment (`///` instead of `//`).

![image](https://user-images.githubusercontent.com/4888530/114220103-6b1d0800-9920-11eb-97d3-44e172a2b91b.png)

Reply:
I tried to find some alternatives.
 * #[profiling::function] makes a massive scope name (`<demo::features::debug3d::extract::Debug3dExtractJob as rafx_framework::nodes::jobs::extract::ExtractJob>::extract`)
 * scope!(std::any::type_name::<self>()) is still long (`demo::features::debug3d::extract::Debug3dExtractJob`)
 * I also tried calling `self.feature_debug_name()` but couldn't find a good way to concat that at compile time

Using super::extract_scope - using super:: does help a lot with knowing where this actually is located. I still don't like it being generated by a macro (feels like magic in a bad way). But if you feel strongly about this over just putting in a string, I'm willing to go ahead with it. But lets name it EXTRACT_SCOPE_NAME.

Reply:
Done.